### ISSUE TITLE
powershell args array is not processed

### ISSUE THREAD

**Issue by mahmoudkassem** [alacritty/alacritty#3375]
Lovely terminal emulator using it on my private Linux laptop and now at work. It seems that the args are not passed to Powershell at all. Passing invalid args does not cause an error message.

OS: Windows
Version: 0.4.1

Complete configuration file:
![alacritty_powershell_not_processing_args](https://user-images.githubusercontent.com/5838197/75146796-17d7fe00-56fc-11ea-85c6-b4829cd7af7f.png)

After start up:
![alacritty_powershell_not_processing_args_1](https://user-images.githubusercontent.com/5838197/75147024-aa789d00-56fc-11ea-8c73-01f0aff71a61.png)

**Comment by chrisduerr** [comment]
Have you tried quoting the arguments? Maybe the `-` is causing yaml some troubles.

Also just to make sure, `powershell -nologo -noprofile` does work, right?

And can you try running `alacritty -e powershell -nologo -noprofile`?

**Comment by enizor** [comment]
I tried quickly on my Win10 machine and the `-nologo` arg did remove the copyright notice, but was not sure how to test the `-noprofile` argument.

>  Passing invalid args does not cause an error message.

It actually gave me an error message when I tried. Maybe alacritty does not read this config file but another? 

I can do some other tests tomorrow if needed.

**Comment by kchibisov** [comment]
>Maybe alacritty does not read this config file but another?

Try to run it with `alacritty -v`, it should print the path to a config file.

**Comment by mahmoudkassem** [comment]
running `alacritty -v` opens a new window instead of showing the config file. It is located at ~\AppData\alacritty\alacritty.yml. The config is applied colors/transparency/font and the copy by selecting and launching URL in browser work too.

I tried the explicit array syntax:
![pass_args_as_array](https://user-images.githubusercontent.com/5838197/75181082-cb151700-573d-11ea-8bef-4f6342b0f2e3.png)

and passing the args in the same line:
![one_line_args](https://user-images.githubusercontent.com/5838197/75181124-e1bb6e00-573d-11ea-8ff0-2a92095aee43.png)

both didn't work, launching with the -e flag in the link did work:
![this_works](https://user-images.githubusercontent.com/5838197/75181206-03b4f080-573e-11ea-804b-f9d8a1fcb1d8.png)
![terminal_with_flags_applied](https://user-images.githubusercontent.com/5838197/75181454-8938a080-573e-11ea-8226-74802cb6232d.png)

**Comment by kchibisov** [comment]
>running alacritty -v opens a new window instead of showing the config file.

It should show a path to a config file in a log, not open it for you.

```
~ ➜ alacritty -v
Created log file at "/tmp/Alacritty-669934.log"
[2020-02-24 21:56:09.201387651] [INFO] Welcome to Alacritty
[2020-02-24 21:56:09.201418602] [INFO] Configuration loaded from "/home/kchibisov/.config/alacritty/alacritty.yml" <- Notice this line
[2020-02-24 21:56:09.244632792] [INFO] Device pixel ratio: 1
[2020-02-24 21:56:09.248004863] [INFO] Initializing glyph cache...

```

**Comment by kchibisov** [comment]
Could someone provide output from the following patch? I can't repro this issue on GNU/Linux with `bash`, so I'd assume it's windows related one.

```diff
diff --git a/alacritty_terminal/src/tty/windows/mod.rs b/alacritty_terminal/src/tty/windows/mod.rs
index 03e2a3c..867af54 100644
--- a/alacritty_terminal/src/tty/windows/mod.rs
+++ b/alacritty_terminal/src/tty/windows/mod.rs
@@ -306,10 +306,12 @@ fn cmdline<C>(config: &Config<C>) -> String {
     let default_shell = Shell::new("powershell");
     let shell = config.shell.as_ref().unwrap_or(&default_shell);

-    once(shell.program.as_ref())
+    let cmdline = once(shell.program.as_ref())
         .chain(shell.args.iter().map(|a| a.as_ref()))
         .collect::<Vec<_>>()
-        .join(" ")
+        .join(" ");
+    println!("{}", cmdline);
+    cmdline
 }
```

**Comment by mahmoudkassem** [comment]
> It should show a path to a config file in a log, not open it for you.

running it on Linux in bash shows the output pointing to the config file, but it also opens a new terminal window.

**Comment by chrisduerr** [comment]
> running it on Linux in bash shows the output pointing to the config file, but it also opens a new terminal window.

Yes, that's the point. It also does this on Windows.

The suggestions like `alacritty -e` and `alacritty -v` should all be executed in a shell, so you can see the output, rather than putting them into some GUI config.

> launching with the -e flag in the link did work

Launching with the `-e` flag executes the exact same codepath as the configuration file. I suggest applying kchibisov's patch which will likely make the error pretty clear.

**Comment by nixpulvis** [comment]
@mahmoudkassem I've tested this on my Windows install, and he `-nologo` does suppress my startup messages without a problem, and `-noprofile` also works as expected. I'm guessing you're not editing the correct configuration file.

> > It should show a path to a config file in a log, not open it for you.

> running it on Linux in bash shows the output pointing to the config file, but it also opens a new terminal window.

Sorry, this was kinda confusing. The point of telling you to run alacrittly with `-v` was to show you the location of your config file. To do this easily just type `alacritty -v` in an instance of alacritty itself, then ignore the window that opens (this is expected, since `-v` just stands for "verbose" logging), and read the line: `... [INFO] Configuration loaded from ...` to find the correct path.

I used this minimal config (this is the entire file) and it worked as intended:

```yml
shell:
 program: powershell
 args:
   - -noprofile
   - -nologo
```

**Comment by mahmoudkassem** [comment]
The issue is related to the config as suggested. I've placed my config file in the ~\AppData\alacritty\alacritty.yml as I'm not too knowledgable with Windows I assumed that %APPDATA%\alacritty\alacritty was actually pointing to AppData. After pasting it in the file explorer it correctly pointed at ~\AppData\Roaming\alacritty\alacritty.yml. It is interesting that there was already a file which had all the settings aside from the shell args setting. Maybe I edited that one initially and got confused afterwards.

`alacritty -v` still doesn't have any output though, but the initial Issue is resolved, the problem was me misconfiguring alacritty. After setting the config in %APPDATA%\alacritty\alacritty.yml it correctly took effect

Thank you all for helping resolve this!! And thank you for this awesome terminal emulator makes the Windows experience much less painful :).
